### PR TITLE
Webpack 4 and production script

### PR DESCRIPTION
**Pull request by lybaokhanh**
Hi @minhnb 

Please help me review this PR.

## What new in this PR:

- Webpack 4
- Babel 7
- React 16.3.2
- Better loading time for development (less than 2 second)
- HMR
- Build script for production
- Express server for production with compression.
- Moving @types to dev dependencies
- And more...

### CODE REVIEW

**Comment by minhnb** [comment]
- Some icon disappear on Login and Register page
- On dev mode, the way website load new "page" is not smooth

**Comment by pvtri96** [comment]
@minhnb 
The unsmooth reload animation is coming from CORE-UI. After removing it, thing is getting smoothly.

File: src/Modules/Station/Components/AddSong/SearchSong/SearchSong.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ export class SearchSong extends Component<any, ISearchSongState> {
     return (
       <div className="search-input">
         <Autosuggest
-          suggestions={suggestions}
+          suggestions={suggestions} // TODO: invalid type of suggestions
```
Comment (minhnb) [inline src/Modules/Station/Components/AddSong/SearchSong/SearchSong.tsx]: why?

File: src/index.tsx [hunk 2]
```diff
@@ -15,11 +15,11 @@ import '../scss/core/_dropdown-menu-right.scss';
 import '../scss/style.scss';
 
 // config redux
+import { configureStore } from 'Configuration/Redux';
 import { Provider } from 'react-redux';
-import { configureStore } from './Configuration/Redux';
```
Comment (minhnb) [inline src/index.tsx]: it should be in another pull request

**Comment by minhnb** [comment]
Please just commit what is related to your task
